src/controllers/activity.controller.ts

```diff
@@ -0,0 +1,306 @@
+                valuePerUnit: 100,
+                unitDescription: 'per post',
+              },
+            },
+          },
+        },
+      },
+      description: '',
+    })
+    activity: Omit<GoodActivity, 'id'>,
+  ): Promise<unknown> {
+    return this.repository.create(activity);
+  }
+
+  /**
+   * Update an Activity's details
+   *
+   * @param id The PoG ID of the activity
+   * @param activity
+   */
+  @put('/activity/{id}', {
+    summary: 'Change Activity Details',
+    operationId: 'put-activity',
+    responses: {
+      '200': {
+        description: 'Activity Updated',
+      },
+      '401': {
+        description: 'Unauthorized',
+        content: {
+          'application/json': {
+            schema: {
+              $ref: '#/components/schemas/ErrorResponse',
+            },
+          },
+        },
+        headers: {},
+      },
+      '404': {
+        description: 'Activity Not Found',
+        content: {
+          'application/json': {
+            schema: {
+              $ref: '#/components/schemas/ErrorResponse',
+            },
+          },
+        },
+      },
+    },
+    requestBody: {
+      content: {
+        'application/json': {
+          schema: {
+            $ref: '#/components/schemas/GoodActivity_Input',
+          },
+          examples: {
+            'Change Activity Details': {
+              value: {
+                name: 'Good Deed Post on Discord - Local Cleanup',
+                goodCategoryId: 8,
+                goodTypeId: 6,
+                valuePerUnit: 100,
+                unitDescription: 'per image',
+              },
+            },
+          },
+        },
+      },
+    },
+    description: "Update an Activity's details",
+    security: [
+      {
+        Activity_API_Key: [],
+      },
+    ],
+    parameters: [
+      {
+        name: 'id',
+        in: 'path',
+        required: true,
+        description: 'The PoG ID of the activity',
+      },
+    ],
+  })
+  async putActivity(
+    @param({
+      schema: {
+        type: 'integer',
+        example: 3,
+      },
+      name: 'id',
+      in: 'path',
+      required: true,
+      description: 'The PoG ID of the activity',
+    })
+    id: number,
+    @requestBody({
+      content: {
+        'application/json': {
+          schema: {
+            $ref: '#/components/schemas/GoodActivity_Input',
+          },
+          examples: {
+            'Change Activity Details': {
+              value: {
+                name: 'Good Deed Post on Discord - Local Cleanup',
+                goodCategoryId: 8,
+                goodTypeId: 6,
+                valuePerUnit: 100,
+                unitDescription: 'per image',
+              },
+            },
+          },
+        },
+      },
+    })
+    activity: GoodActivityInput | any,
+  ): Promise<unknown> {
+    delete activity?.id;
+    return this.repository.updateById(id, activity).catch((err: Error) => {
+      if (err.message == '404')
```

Comment:
```suggestion
      if (err.message == 'Document not found')
```

Reply:
which means L169 of src\connectors\BetterFirestoreConnector.ts needs to be changed